### PR TITLE
Add Windows support and set up cargo-dist builds

### PR DESCRIPTION
**Pull request by shadaj**
Set up initial cargo-dist builds

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        include:
+        - os: ubuntu-20.04
+          deps-script: sudo apt-get install -y protobuf-compiler
+          dist-args: --artifacts=global
+          install-dist: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/axodotdev/cargo-dist/releases/download/v0.0.7/cargo-dist-installer.sh | sh
+        # - os: macos-11
+        #   deps-script: brew install protobuf
+        #   dist-args: --artifacts=local --target=aarch64-apple-darwin --target=x86_64-apple-darwin
+        #   install-dist: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/axodotdev/cargo-dist/releases/download/v0.0.7/cargo-dist-installer.sh | sh
+        - os: ubuntu-20.04
```
Comment (ramnivas) [inline .github/workflows/build.yml:23]: Is the duplicated `os: ubuntu-20.04` intended?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        include:
+        - os: ubuntu-20.04
+          deps-script: sudo apt-get install -y protobuf-compiler
+          dist-args: --artifacts=global
+          install-dist: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/axodotdev/cargo-dist/releases/download/v0.0.7/cargo-dist-installer.sh | sh
+        # - os: macos-11
+        #   deps-script: brew install protobuf
+        #   dist-args: --artifacts=local --target=aarch64-apple-darwin --target=x86_64-apple-darwin
+        #   install-dist: curl --proto '=https' --tlsv1.2 -LsSf https://github.com/axodotdev/cargo-dist/releases/download/v0.0.7/cargo-dist-installer.sh | sh
+        - os: ubuntu-20.04
```
Comment (ramnivas) [inline .github/workflows/release.yml:94]: Duplicate `os: ubuntu-20.04`

File: Cargo.toml [hunk 3]
```diff
@@ -82,7 +82,30 @@ wildmatch = "2.1.0"
 [profile.dev]
 debug = 1
 
+# Config for 'cargo dist'
+[workspace.metadata.dist]
+# The preferred cargo-dist version to use in CI (Cargo.toml SemVer syntax)
+cargo-dist-version = "0.0.7"
+# The preferred Rust toolchain to use in CI (rustup toolchain syntax)
+rust-toolchain-version = "1.67.1"
```
Comment (ramnivas) [inline Cargo.toml:90]: How does this relates to the version we specify in `rust-toolchain.toml`?

**Review by ramnivas — APPROVED** [review]

**Review by ramnivas — COMMENTED** [review]
Other than some duplicate, looks great.

File: .github/workflows/build.yml (hunk 1 above)
Comment (ramnivas) [reply .github/workflows/build.yml:23]: I see one is marked as `--artifacts=global` and other `--artifacts=local --target=x86_64-unknown-linux-gnu`

File: .github/workflows/release.yml (hunk 2 above)
Comment (shadaj) [reply .github/workflows/release.yml:94]: These are one per matrix

File: .github/workflows/build.yml (hunk 1 above)
Comment (shadaj) [reply .github/workflows/build.yml:23]: The global one is the one that generates additional release metadata

**Review by ramnivas — APPROVED** [review]

**Pull request merged**